### PR TITLE
Records the maximum frequency for both actual data and halo output.

### PR DESCRIPTION
**Pull request by matthewclegg**

### CODE REVIEW

File: src/models/gamma_poisson_model.py [hunk 1]
```diff
@@ -131,6 +131,7 @@ def __init__(
             the inventory is at least twice the number of impressions that were
             observed in the campaign.
         """
+        super().__init__(data, max_reach)
```
Comment (jiayu-google) [inline src/models/gamma_poisson_model.py:134]: This line is Ok but not useful?

`self._data` seems not used anywhere.   `self._max_reach` has been specified in line #144 below.

File: src/models/goerg_model.py [hunk 2]
```diff
@@ -37,6 +37,7 @@ def __init__(self, data: [ReachPoint]):
           max_reach:  Optional.  If specified, the maximum possible reach that can
             be achieved.
         """
+        super().__init__(data)
```
Comment (jiayu-google) [inline src/models/goerg_model.py:40]: Ditto

File: src/models/tests/reach_curve_test.py [hunk 3]
```diff
@@ -44,20 +44,24 @@ def by_spend(self, spend: [float], max_frequency: int = 1) -> ReachPoint:
 class ReachCurveTest(absltest.TestCase):
     def test_reach(self):
         model = FakeReachCurve([ReachPoint([200], (100,))])
+        model._fit()
```
Comment (jiayu-google) [inline src/models/tests/reach_curve_test.py:47]: Note that FakeReachCurve belongs to ReachCurves which belongs to ReachSurface.
`ReachSurface` has done `_fit()` in `__init__()`.  See line 42 of https://github.com/world-federation-of-advertisers/planning-evaluation-framework/blob/abe1c96ec93cf4d6c2d04808d8211c7c770739d2/src/models/reach_surface.py#L24

So this line is ok but not necessary?

**Review by jiayu-google — APPROVED** [review]
LGTM but left minor comments about possibly redundant codes.

File: src/models/gamma_poisson_model.py (hunk 1 above)
Comment (matthewclegg) [reply src/models/gamma_poisson_model.py:134]: See `src/driver/experimental_trial.py`, line 300.

File: src/models/goerg_model.py (hunk 2 above)
Comment (matthewclegg) [reply src/models/goerg_model.py:40]: See src/driver/experimental_trial.py, line 300.

File: src/models/tests/reach_curve_test.py (hunk 3 above)
Comment (matthewclegg) [reply src/models/tests/reach_curve_test.py:47]: Line 42 of `reach_surface.py` is deleted in this PR.  After this PR, it is necessary to always explicitly call `_fit()` after a model is instantiated.

**Review by RonPinelliMRC — APPROVED** [review]
MRC approving based on cursory review after confirming non-independent code review on an exception basis. Experimental code will require independent review prior to publication of or reliance on results generated.

**Pull request merged**